server/tidb_test.go

```diff
@@ -131,3 +132,14 @@ func (ts *TidbTestSuite) TestSocket(c *C) {
 	dsn = tcpDsn
 	server.Close()
 }
+
+func (ts *TidbTestSuite) TestIssue3662(c *C) {
+	db, _ := sql.Open("mysql", "root@tcp(localhost:4001)/a_database_not_exist")
+	defer db.Close()
+	// According to documentation, "Open may just validate its arguments without
+	// creating a connection to the database. To verify that the data source name
+	// is valid, call Ping."
+	err := db.Ping()
```

Comment:
But I'm not quite understand the purpose of  this testing.
Ping should send a `mysql.ComPing`, which is handled in function `dispatch` in conn.go, but if handshake fail, it would never run to there.

What happen if change it to 
```
sql.Open("mysql", "root@tcp(localhost:4001)/a_database_exist")
```